README.md

```diff
@@ -1 +1,91 @@
-# node-non-blocking-callback-lab
\ No newline at end of file
+# Callback Hell Refactoring Lab
+
+## Objectives
+
+1. Refactor an async code using named functions
+1. Refactor an async code using modules
+
+## Introduction
+
+You've been tasked by your dev team to refactor some ugly code. You colleagues know you are the master of simplifying code, and dealing with callback hell.
+
+The code is converting private information from a CSV format into a JSON with obfuscated credit card numbers. Security is important. That's why this script is hiding all the credit card number digits except the last four. We don't want hackers to get the CC numbers so easily if the data gets compromised!
+ 
+In this lab, you'll start with an ugly, huge and complex looking file, refactor code using named functions and then abstract the code into 3 modules. 
+
+## Instructions
+
+1. Open `mask.js` file. Understand how it's working by looking at the code.
```

Comment:
This is both very specific and very daunting, especially because `mask.js` is intentionally in need of refactoring.

More than that, it's difficult to understand code just by looking at -- I would encourage students to run the code, to use the debugger, etc.